DelvEdit/src/com/interrupt/dungeoneer/editor/file/EditorFile.java

```diff
@@ -269,58 +271,95 @@ public boolean result(boolean success, FileHandle result) {
-                    openLevel.loadForEditor(dir + fileName, heightFile);
-                }
-                else if(fileName.endsWith(".bin")) {
-                    openLevel = KryoSerializer.loadLevel(levelFileHandle);
-                    openLevel.init(Level.Source.EDITOR);
-                }
-                else {
-                    openLevel = Game.fromJson(Level.class, levelFileHandle);
-                    openLevel.init(Level.Source.EDITOR);
+    private void openInternal(FileHandle file) {
+        file = Gdx.files.getFileHandle(file.file().getAbsolutePath(), Files.FileType.Absolute);
+        String name = file.name();
+
+        if (file.exists()) {
+            Level level;
+
+            try {
+                if (name.endsWith(".png")) {
+                    level = loadPngFile(file);
+                } else if (name.endsWith(".bin")) {
+                    level = loadBinFile(file);
+                } else if (name.endsWith(".dat")) {
+                    level = loadDatFile(file);
+                } else {
+                    showWarningDialog("Unkown extension. Cannot load '" + name + "'.");
+                    return;
                 }
+            } catch (SerializationException exception) {
+                showWarningDialog("Corrupted file data. Cannot load '" + file.name() + "'.");
+                return;
+            } catch (Exception exception) {
+                showWarningDialog(exception.getMessage() + " Cannot load '" + file.name() + "'.");
+                return;
+            }
 
-                Editor.app.level = openLevel;
-                Editor.app.refresh();
-                Editor.app.cameraController.setPosition(openLevel.width / 2f, 4.5f, openLevel.height / 2f);
+            if (!validateLevel(level)) {
+                showWarningDialog("Invalid level data. Cannot load '" + file.name() + "'.");
+                return;
+            }
 
-                Editor.app.history = new EditorHistory();
-                Editor.app.history.saveState(Editor.app.level);
-                Editor.app.file.markClean();
+            Editor.app.file = new EditorFile(file);
 
-                Editor.options.recentlyOpenedFiles.removeValue(levelFileHandle.path(), false);
-                Editor.options.recentlyOpenedFiles.insert(0, levelFileHandle.path());
+            Editor.app.level = level;
+            Editor.app.refresh();
+            Editor.app.cameraController.setPosition(level.width / 2f, 4.5f, level.height / 2f);
 
-                Editor.app.viewSelected();
-            }
+            Editor.app.history = new EditorHistory();
+            Editor.app.history.saveState(Editor.app.level);
+            Editor.app.file.markClean();
+
+            Editor.options.addRecentlyOpenedFile(file.path());
+
+            Editor.app.clearEntitySelection();
+            Editor.app.viewSelected();
+        } else {
+            showWarningDialog("File does not exist. Cannot load '" + name + "'.");
```

Comment:
A nice improvement would be to remove the file from recently open file list. Probably `Editor.options.removeRecentlyOpenedFile(file.path())`

Reply:
Yes, true. In case that file does not exist anymore, we can remove it from the list.